src/client.rs

```diff
@@ -62,46 +148,84 @@ enum VisibleCellState {
     BlownMine,
 }
 trait TClientField {
+    fn width(&self) -> usize;
```

Comment:
if it is a window size in pixels, then should be renamed to window_width, but in the RectField u fill it with num of cells in a row, which should not be accessible with trait.

the only time u use it is to figure out the window size, so probably just rename (same with heigth)